zuul-core/src/main/java/com/netflix/zuul/netty/filter/BaseZuulFilterRunner.java

```diff
@@ -474,7 +475,12 @@ public void onError(Throwable ex) {
         public void onCompleted() {
             try (TaskCloseable ignored = PerfMark.traceTask(filter, f -> f.filterName() + ".onCompletedAsync")) {
                 PerfMark.linkIn(onCompletedLinkOut.get());
+                if (outMesg == null) {
```

Comment:
And I think this is a non-issue, but does resuming `onCompleted` affect the mutation order of `outMesg`?

Reply:
I don't think there should be any issues. After reviewing things again I did end up adding error handling to onComplete, as the request would hang if an exception was thrown for some reason